wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -782,39 +779,6 @@ public void broadcastTransaction(final Transaction tx) {
         intent.putExtra(BlockchainService.ACTION_BROADCAST_TRANSACTION_HASH, tx.getHash().getBytes());
         startService(intent);
     }
-
-    public static PackageInfo packageInfoFromContext(final Context context) {
-        try {
-            return context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
-        } catch (final NameNotFoundException x) {
-            throw new RuntimeException(x);
-        }
-    }
-
-    public PackageInfo packageInfo() {
-        return packageInfo;
-    }
-
-    public final String applicationPackageFlavor() {
-        final String packageName = getPackageName();
-        final int index = packageName.lastIndexOf('_');
-
-        if (index != -1)
-            return packageName.substring(index + 1);
-        else
-            return null;
-    }
-
-    public static String httpUserAgent(final String versionName) {
-        final VersionMessage versionMessage = new VersionMessage(Constants.NETWORK_PARAMETERS, 0);
-        versionMessage.appendToSubVer(Constants.USER_AGENT, versionName, null);
-        return versionMessage.subVer;
-    }
-
-    public String httpUserAgent() {
-        return httpUserAgent(packageInfo().versionName);
-    }
-
```

Comment:
In the ongoing effort to remove dependencies on the `WalletApplication`, I moved this code into a separate service class.